src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -76,4 +78,56 @@ private[actions] abstract class RefreshActionBase(
           s"Current index state is ${previousIndexLogEntry.state}")
     }
   }
+
+  /**
+   * Compare list of source data files from previous IndexLogEntry to list
+   * of current source data files, validate fileInfo for existing files and
+   * identify deleted source data files.
+   */
+  protected lazy val deletedFiles: Seq[String] = {
+    val rels = previousIndexLogEntry.relations
+    val originalFiles = rels.head.data.properties.content.fileInfos
+    val currentFiles = rels.head.rootPaths
+      .flatMap { p =>
+        Content
+          .fromDirectory(path = new Path(p), throwIfNotExists = true)
+          .fileInfos
+      }
+      .map(f => f.name -> f)
+      .toMap
+
+    var delFiles = Seq[String]()
+    originalFiles.foreach { f =>
+      currentFiles.get(f.name) match {
+        case Some(v) =>
+          if (!f.equals(v)) {
+            throw HyperspaceException(
+              "Index refresh (to handle deleted source data) aborted. " +
+                s"Existing source data file info is changed (file: ${f.name}).")
+          }
+        case None => delFiles :+= f.name
+      }
+    }
+
+    delFiles ++ previousIndexLogEntry.deletedFiles
```

Comment:
Do we need to append previously collected deleted files?

Reply:
`RefreshDeleteAction` can use this complete collection to handle all deleted files.
`RefreshAppendAction` can use this to update the metadata.
So yes. 

Please correct me if I am wrong.

Reply:
So, this is actually a bug fix right (if we had this field populated)?

If so, can you update the comment why we need to add this (e.g., `deletedFiles` is present only if index is not updated to include this info)?

Reply:
Also, I guess we cannot add this to the test using the existing APIs due to missing "quick". If so, can you add a TODO so can follow up when we add a "quick" mode?

Reply:
thanks, added #195